### ISSUE TITLE
VTODO support

### ISSUE THREAD

**Issue by christian-esken** [caldav4j/caldav4j#103]
## Reproduce

Use the caldav4j library
Steps to reproduce the issue:

Version: com.github.caldav4j:caldav4j:0.9.2
Added dependencies:
    implementation 'com.trivago:triava:2.0.1'
    implementation 'javax.servlet:javax.servlet-api:3.1.0'

Server: OwnCloud with Calendar extension
Code: https://github.com/christian-esken/todo4you/blob/master/app/src/main/java/de/todo4you/todo4you/caldav/VTodoCalDAVCollection.java
    
## I expect

caldav4j to support VTODO entities. Operations: GET. ADD. MODIFY (Hint: these may not be offical DAV terms, but it should be clear what they mean. In SQL terms;  SELECT, INSERT, UPDATE).

## Instead
MODIFY is not supported, as far as I saw in examples, docs and browsing source code. GET and ADD is possible.

- GET:  using CalDAVCollection and GenerateQuery. This looks like a clean and intended usage. See get() in https://github.com/christian-esken/todo4you/blob/master/app/src/main/java/de/todo4you/todo4you/caldav/CalDavConnectorHC3.java
- ADD: Works, using CalDAVCollection.add(). I am not sure whether adding VTODO's from that class is officially supported, but it works. See add() in https://github.com/christian-esken/todo4you/blob/master/app/src/main/java/de/todo4you/todo4you/caldav/CalDavConnectorHC3.java
- MODIFY. Not supported as far as I see. CalDAVCollection can modify VEVENT, but not VTODO. I have worked around, subclassing CalDAVCollection : https://github.com/christian-esken/todo4you/blob/master/app/src/main/java/de/todo4you/todo4you/caldav/VTodoCalDAVCollection.java

## Notes

This ticket can be treated as a question. What I would like to know:
- Am I correct that caldav4j does not explicitly support modification of VTODO?
- Is there a lower-level API that I could use?
- Is my solution in  https://github.com/christian-esken/todo4you/blob/master/app/src/main/java/de/todo4you/todo4you/caldav/VTodoCalDAVCollection.java sensible? I needed to copy some methods due to access issues (package, private) or just modified them slightly (VEVENT => VTODO).

**Comment by ioggstream** [comment]
Hi @christian-esken,

can you please add an HTTP TRACE of the error?

**Comment by TheAntimist** [comment]
From what I see, VTODO should already be supported in CalDAVCollection, based on the current release, we only utilize Calendar for the main object. You can choose to utilize VTODO component if you wish to do so. If there is something where we limit to VEVENT do let us know.

We do have lower level api's under the [methods](https://github.com/caldav4j/caldav4j/tree/master/src/main/java/com/github/caldav4j/methods) if you wish to utilize one as such.

**Comment by christian-esken** [comment]
There are some hardcoded VEVENT constatnts used in CalDAVCollection, so I worked around in changing them to  VTODO in my VTodoCalDAVCollection.java. I willl give detailt soon.  I feel petty overloaded curreently, so let me check firtst and I will give feedback in a weekk

**Comment by ioggstream** [comment]
Once we remove deprecated methods, we should probably generalize getMasterEvent -> getMasterComponent.

@christian-esken  PR and associated tests (possibily small, incremental PR since we're now using circleci for CI) are welcome for both ;)

**Comment by TheAntimist** [comment]
Sounds good, will deal with those soon.

**Comment by christian-esken** [comment]
@ioggstream The methods are:
- updateMasterEvent()
- getMasterEvent()

I am refering the 0.9x version. If I create a PR, should I directly head for the master branch? I do not see a branch for 0.9x. So I would just fork, work on master and do a PR, right.

**Comment by TheAntimist** [comment]
I agree with @ioggstream, our development is now focused on Master, which is currently based on Apache HttpClient 4 (1.0.0-rc.2 on maven) . As mentioned some API changes do require you to update your app.
